x-pack/platform/packages/shared/onechat/onechat-common/tools/tool_ids.test.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { toolIdRegexp } from './tool_ids';
+
+describe('toolId regexp', () => {
+  const validToolIds = [
+    'mytool',
+    'core.mytool',
+    'core.foo.mytool',
+    'a',
+    'a.b',
+    'tool_1',
+    'tool-1',
+    'foo_bar-baz.qux_123',
+    'a1.b2.c3',
+    'abc.def_ghi-jkl.mno',
+  ];
+
+  const invalidToolIds = [
```

Comment:
could we also have a invalid tool id test on > 64 characters?

Reply:
yeah forgot about that part of the validation. Will add + add tests